addons/stock_landed_costs/models/stock_landed_cost.py

```diff
@@ -98,13 +93,16 @@ def button_validate(self):
             })
             for line in cost.valuation_adjustment_lines.filtered(lambda line: line.move_id):
                 cost_to_add = line.additional_landed_cost
-                line.move_id.landed_cost_value += cost_to_add
-                
-#                 for quant in line.move_id.quant_ids:
-#                     if quant.location_id.usage != 'internal':
-#                         qty_out += quant.qty
-                line._create_accounting_entries(move, 0) #TODO: qty_out like you would do with the different moves
-                
+                old_value = line.move_id.product_id.stock_value
+                line.move_id.write({'landed_cost_value': line.move_id.landed_cost_value + cost_to_add,
+                                    'value': line.move_id.value + cost_to_add, 
+                                    'price_unit': (line.move_id.value + cost_to_add) / line.move_id.product_qty,})
+                line.move_id.replay()
+                line.move_id.product_id._compute_stock_value()
+                new_value = line.move_id.product_id.stock_value
+                diff = new_value - old_value
+                extra_cost = diff - cost_to_add
+                line._create_accounting_entries(move, diff, extra_cost)
```

Comment:
Landed cost is only to be applied on incoming moves,

No validation is performed when computing or when validating the Landed Costs.

https://github.com/Vauxoo/addons-vauxoo/blob/8.0/stock_landed_costs_average/model/stock_landed_costs.py#L432-L434